### ISSUE TITLE
Validation for strings that have more than just white space.

### ISSUE THREAD

**Issue by RichardMeadowsTC** [PowerShell/PowerShell#10010]
# ValidateNotNullOrEmpty is not sufficient in all cases

Built in function Test-Path will return $true when -Path is just white space.
> Test-Path -Path '   '
Returns $true
Expect it to write-error since that is its what it does when given $null.

I need to validate that my strings are not just white space. Can likely to that with Regx.  Would be useful if it were built in.

# Proposed technical implementation details (optional)
(-not [string]::IsNullOrWhiteSpace($_))

Would like to see a no null or white space validation feature.
    [Parameter(Mandatory)]
    [ValidateNotNullOrWhiteSpace()]
    [string] $Path,

Such that my function fails to be called when $Path is just white space, $null, or empty string.

**Comment by vexx32** [comment]
Rather than create what is essentially a duplicate attribute, why not just have a settable property on the existing attribute to also check for whitespace? For example:

```powershell
[ValidateNotNullOrEmpty(RejectWhitespace)]
```

**Comment by BrucePay** [comment]
Note - the specific issue with `Test-Path` appears to be fixed in 6.2.0
```powershell
PSCore (1:2) >  Test-Path -Path ' '
False
```

As far as checking for whitespace in an argument,, you could use the `ValidatePattern` attribute with something like `[validatepattern("^[^ ]*$")]`.

**Comment by vexx32** [comment]
Yeah, I fixed that one. It bothered me _far_ too much. 😄 

The only way to get `Test-Path` to throw now is to actually pass it a `$null` value.
